xbmc/platform/darwin/ios-common/DarwinEmbedNowPlayingInfoManager.mm

```diff
@@ -49,8 +49,21 @@ - (void)onPlay:(NSDictionary*)item
   {
     if (auto image = [UIImage imageWithContentsOfFile:thumb])
     {
+      auto square_size = image.size.height;
+      if (image.size.width > square_size)
+      {
+        square_size = image.size.width;
+      }
+      UIImageView* view =
+          [[UIImageView alloc] initWithFrame:CGRectMake(0, 0, square_size, square_size)];
+      [view setImage:image];
+      view.contentMode = UIViewContentModeScaleAspectFit;
```

Comment:
don't create dummy view for this, use `AVMakeRectWithAspectRatioInsideRect` function to get image drawing rect instead. And then just call `[image drawInRect:]`.

Reply:
I guess should be this:
```objc
auto rect = CGRectMake(0, 0, square_size, square_size);
UIGraphicsBeginImageContextWithOptions(rect.size, YES, 0.0);
[image drawInRect:AVMakeRectWithAspectRatioInsideRect(image.size, rect)];
image = UIGraphicsGetImageFromCurrentImageContext();
UIGraphicsEndImageContext();
```

Reply:
Thank you for your help :-)

Reply:
I hope you have runtime-tested the change? Because I'm not 100% sure :)

Reply:
Yes I did it, it works ;-)